pkg/clusteragent/autoscaling/externalmetrics/metrics_retriever.go

```diff
@@ -19,11 +19,11 @@ import (
 
 const (
 	invalidMetricErrorMessage                  string = "%v, query was: %s"
-	invalidMetricErrorWithRetriesMessage       string = "%v, query was: %s, will retry after %s."
+	invalidMetricErrorWithRetriesMessage       string = "%v, query was: %s, will retry after %s"
 	invalidMetricOutdatedErrorMessage          string = "Query returned outdated result, check MaxAge setting, query: %s"
-	invalidMetricNotFoundErrorMessage          string = "Unexpected error, query data not found in result, query: %s"
-	invalidMetricGlobalErrorMessage            string = "Global error (all queries) from backend, invalid syntax in query? Check Cluster Agent leader logs for details"
-	invalidMetricGlobalErrorWithRetriesMessage string = "Global error (all queries, batch size %d) from backend, invalid syntax in query? Check Cluster Agent leader logs for details. Will retry after %s."
+	invalidMetricNotFoundErrorMessage          string = "Unexpected error, query data not found in result, query: %s. Query error: %v"
```

Comment:
I'm not sure sure if we want to leak internal errors from API in the error message. You may simply add dedicated error message based on error returned by the backend.